domain/application/service/application.go

```diff
@@ -311,18 +316,28 @@ func makeUpsertUnitArgs(in AddUnitArg) application.UpsertUnitArg {
 			// TODO(units) - handle the in.CloudContainer.Address space ID
 			// For k8s we'll initially create a /32 subnet off the container address
 			// and add that to the default space.
-			result.CloudContainer.Address = &application.Address{
+			result.CloudContainer.Address = &application.ContainerAddress{
+				// For cloud containers, the device is a placeholder without
+				// a MAC address and once inserted, not updated. It just exits
+				// to tie the address to the net node corresponding to the
+				// cloud container.
+				Device: application.ContainerDevice{
```

Comment:
It will be implicitly used when addresses are retrieved. There's a test which validates addresses are created as expected using this SQL
```
SELECT address_value, type_id, scope_id, origin_id FROM ip_address ipa
JOIN link_layer_device lld ON lld.uuid = ipa.device_uuid
JOIN unit u ON u.net_node_uuid = lld.net_node_uuid WHERE u.name = ?
```